cri/v1alpha1/cri.go

```diff
@@ -710,11 +710,16 @@ func (c *CriManager) ContainerStatus(ctx context.Context, r *runtime.ContainerSt
 
 	labels, annotations := extractLabels(container.Config.Labels)
 
-	imageRef := container.Image
-	imageInfo, err := c.ImageMgr.GetImage(ctx, imageRef)
+	// FIXME(fuwei): if user repush image with the same reference, the image
+	// ID will be changed. For now, pouch daemon will remove the old image ID
+	// so that CRI fails to fetch the running container. Before upgrade
+	// pouch daemon image manager, we use reference to get image instead of
+	// id.
+	imageInfo, err := c.ImageMgr.GetImage(ctx, container.Config.Image)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get image %s: %v", imageRef, err)
+		return nil, fmt.Errorf("failed to get image %s: %v", container.Config.Image, err)
 	}
+	imageRef := imageInfo.ID
```

Comment:
Emmm, I think we could directly use `imageID` variable name here.

Reply:
If user has changed Dockerfile and re-builds it with the same reference, the imageID will be changed. This is tricky part that we use reference to `GetImage` to make sure that `ContainerStatus` can be successfully executed after imageID was changed. That is why that we should use `imageInfo.ID` here.

Reply:
I know what you mean, what I want to say is should we use `imageID := imageInfo.ID` here to avoid the misunderstanding?

Reply:
CRI defines the `imageRef` in the protobuf. we just follow the rule here.